src/terminal/parser/InputStateMachineEngine.cpp

```diff
@@ -395,6 +386,26 @@ bool InputStateMachineEngine::ActionCsiDispatch(const wchar_t wch,
     const auto remainingArgs = parameters.size() > 1 ? parameters.substr(1) : std::basic_string_view<size_t>{};
 
     bool success = false;
+    // Handle intermediate characters, if any
+    if (!intermediates.empty())
+    {
+        switch (static_cast<CsiIntermediateCodes>(intermediates.at(0)))
+        {
+        case CsiIntermediateCodes::MOUSE_SGR:
+        {
+            DWORD buttonState = 0;
+            DWORD eventFlags = 0;
+            modifierState = _GetSGRMouseModifierState(parameters);
+            success = _GetSGRXYPosition(parameters, row, col);
+            success = _UpdateSGRMouseButtonState(wch, parameters, buttonState, eventFlags) && success;

```

Comment:
I mean it's honestly weird that you put the `success` on the right side of this one and the left half of the one below. It should always be on the left so it can short-circuit.

Reply:
Leaving a comment:
```
// we need _UpdateSGRMouseButtonState() on the left side here because we _always_ should be updating our state
// even if we failed to parse a portion of this sequence.
```